crates/services/gas_price_service/src/fuel_gas_price_updater/da_source_adapter/service.rs

```diff
@@ -13,39 +11,93 @@ use std::{
     time::Duration,
 };
 use tokio::{
-    sync::mpsc::Sender,
+    sync::broadcast::Sender,
     time::{
         interval,
         Interval,
     },
 };
 
 pub use anyhow::Result;
+use tokio::sync::broadcast::Receiver;
+
+pub struct DaBlockCostsProvider<Source: DaBlockCostsSource + 'static> {
+    handle: ServiceRunner<DaBlockCostsService<Source>>,
+    subscription: Receiver<DaBlockCosts>,
+}
+
+#[async_trait::async_trait]
+pub trait DaBlockCostsProviderPort {
```

Comment:
I don't see the reason of having trait for `DaBlockCostsProvider`. It looks like we can just implement methods directly on the `DaBlockCostsProvider`.

Reply:
addressed in d68b02d83